### PR TITLE
docs: revamp the README

### PR DESCRIPTION
**Pull request by rexledesma**
### Summary & Motivation
Took a lot of inspiration from https://github.com/facebook/react/blob/main/README.md. 

We have a lot of relevant links for people to check out when they first discover Dagster. We should focus on providing resources for first steps and subsequent steps to explore and join the community. It should be short and to the point, enough to whet people's appetite. I've added these and other specific actionable steps. 

I removed our list of integrations because that's already covered in our documentation. Also it took up like half the README. 

### How I Tested These Changes
https://github.com/dagster-io/dagster/tree/rl/05-24-docs_revamp_the_README#readme

### CODE REVIEW

**Comment by rexledesma** [comment]
Current dependencies on/for this PR:
* master
  * **PR #8052** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/8052" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/8052?utm_source=stack-comment).

File: README.md [hunk 1]
```diff
@@ -50,112 +42,41 @@ $ pip install dagster dagit
...
+- **Dagit**: The web interface for developing and operating Dagster jobs. It includes a DAG browser,
+a type-aware interface to launch runs, a live view for in-progress runs, a catalog to view your data
+assets, and more.
 
-### Learn
+For a quick overview, check out our [Getting Started](https://docs.dagster.io/getting-started) page.
 
-Next, jump right into our [tutorial](https://docs.dagster.io/tutorial/), read our [complete documentation](https://docs.dagster.io), or check out our [GitHub Discussions](https://github.com/dagster-io/dagster/discussions). If you're actively using Dagster or have questions on
-getting started, we'd love to hear from you:
+### Documentation
```
Comment (yuhan) [inline README.md:50]:
```suggestion
## Documentation
```

File: README.md [hunk 2]
```diff
@@ -50,112 +42,41 @@ $ pip install dagster dagit
...
+
+We've divided up the documentation into several sections:
+
+- [🌱 Tutorial](https://docs.dagster.io/tutorial/)
+- [💡 Concepts](https://docs.dagster.io/concepts)
+- [🚢 Deployment](https://docs.dagster.io/deployment)
+- [🤝 Integrations](https://docs.dagster.io/integrations)
+- [📖 Guides](https://docs.dagster.io/guides)
+
+### Community
```
Comment (yuhan) [inline README.md:62]:
```suggestion
## Community
```

File: README.md [hunk 3]
```diff
@@ -50,112 +42,41 @@ $ pip install dagster dagit
...
-  single-process and multi-process execution engines; and a CLI tool for driving those engines.
-- **Dagit**: the UI for developing and operating Dagster pipelines, including a DAG browser, a
-  type-aware config editor, and a live execution interface.
+- **Dagster**: The core programming model.
+- **Dagit**: The web interface for developing and operating Dagster jobs. It includes a DAG browser,
+a type-aware interface to launch runs, a live view for in-progress runs, a catalog to view your data
+assets, and more.
 
-### Learn
+For a quick overview, check out our [Getting Started](https://docs.dagster.io/getting-started) page.
```
Comment (yuhan) [inline README.md:48]: I think we should have an "Example" or "Quick Start" section here to show code snippets as opposed to redirecting folks to docs site - cc @sryza / @erinkcochran87 do you have thoughts on what code to include as the starter code? - we can default to the same thing as the [Getting Started](https://docs.dagster.io/getting-started#writing-a-job) for now and switch to assets content later.

Comment (erinkcochran87) [reply README.md:48]: If we went the route of it actually being a quick start, I could be on board with it. I feel like including examples would require additional explanation that's typically too lengthy for a README.

I think it'd also be cool to add something that's similar to what [Jekyll does here](https://github.com/jekyll/jekyll#getting-started), specifically the **Getting started** and **Diving in** sections. After someone does the quick start, I think pointing them in the direction of next steps would be helpful. (FWIW, I also think we should do this in the docs - working on that 😉  )

File: README.md [hunk 4]
```diff
@@ -8,39 +8,29 @@
...
-### Debug pipelines from a rich UI
-
-Dagit, Dagster’s web interface, includes expansive facilities for understanding the jobs it orchestrates. When inspecting a run of your job, you can query over logs, discover the most time consuming tasks via a Gantt chart, re-execute subsets of steps, and more.
-
-## Getting Started
-
-### Installation
+Dagster is an orchestration platform for the development, production, and observation of data assets.
+
+- **Develop and Test Locally, then Deploy Anywhere:** With Dagster, the same computations can run
```
Comment (erinkcochran87) [inline README.md:13]:
```suggestion
- **Develop and test locally, then deploy anywhere:** With Dagster, the same computations can run
```

Let's keep to sentence case here for consistency.

File: README.md [hunk 5]
```diff
@@ -8,39 +8,29 @@
...
+in-process against your local file system, or on a distributed work queue against your production
```
Comment (erinkcochran87) [inline README.md:14]:
```suggestion
in-process against your local file system or on a distributed work queue against your production
```

Removing unnecessary comma.

File: README.md [hunk 6]
```diff
@@ -8,39 +8,29 @@
...
+data lake. You can choose to locally develop on your labtop, deploy on-premise, or run in any
+cloud.
+- **Model the Data Produced and Consumed:** In your orchestration graph, Dagster models data
```
Comment (erinkcochran87) [inline README.md:17]:
```suggestion
- **Model the data produced and consumed:** In your orchestration graph, Dagster models data
```

File: README.md [hunk 7]
```diff
@@ -8,39 +8,29 @@
...
+dependencies and handles how data passes between steps. Gradual typing on inputs and outputs catches
+bugs early.
+- **Link Data To Computations:** Dagster’s Asset Catalog tracks the data sets and ML models produced
```
Comment (erinkcochran87) [inline README.md:20]:
```suggestion
- **Link data to computations:** Dagster’s Asset Catalog tracks the data sets and ML models produced
```

File: README.md [hunk 8]
```diff
@@ -8,39 +8,29 @@
...
+by your jobs. You can understand how they were generated and trace issues when asset declarations
```
Comment (erinkcochran87) [inline README.md:21]:
```suggestion
Understand how they were generated and trace issues when asset declarations
```

Understood subject here, too.

File: README.md [hunk 9]
```diff
@@ -8,39 +8,29 @@
...
+do not match their materializations in storage.
+- **Build a Self-Service Data Platform:** Dagster helps platform teams build systems for data
```
Comment (erinkcochran87) [inline README.md:23]:
```suggestion
- **Build a self-service data platform:** Dagster helps platform teams build systems for data
```

File: README.md [hunk 10]
```diff
@@ -8,39 +8,29 @@
...
-Dagit, Dagster’s web interface, includes expansive facilities for understanding the jobs it orchestrates. When inspecting a run of your job, you can query over logs, discover the most time consuming tasks via a Gantt chart, re-execute subsets of steps, and more.
-
-## Getting Started
-
-### Installation
+Dagster is an orchestration platform for the development, production, and observation of data assets.
+
+- **Develop and Test Locally, then Deploy Anywhere:** With Dagster, the same computations can run
+in-process against your local file system, or on a distributed work queue against your production
+data lake. You can choose to locally develop on your labtop, deploy on-premise, or run in any
```
Comment (erinkcochran87) [inline README.md:15]:
```suggestion
data lake. Choose to locally develop on your laptop, deploy on-premise, or run in any
```

I vote for using an understood subject for conciseness. It also makes this sentence sound more compelling.

File: README.md [hunk 11]
```diff
@@ -8,39 +8,29 @@
...
+- **Model the Data Produced and Consumed:** In your orchestration graph, Dagster models data
+dependencies and handles how data passes between steps. Gradual typing on inputs and outputs catches
+bugs early.
+- **Link Data To Computations:** Dagster’s Asset Catalog tracks the data sets and ML models produced
+by your jobs. You can understand how they were generated and trace issues when asset declarations
+do not match their materializations in storage.
+- **Build a Self-Service Data Platform:** Dagster helps platform teams build systems for data
+practitioners. Jobs are built from shared, reusable, configurable data processing components.
+Dagit, Dagster’s web interface, lets anyone inspect these objects and discover how to use them.
+- **Declare and Isolate Dependencies:** Dagster’s server model lets you isolate codebases. Problems
```
Comment (erinkcochran87) [inline README.md:26]:
```suggestion
- **Declare and isolate dependencies:** Dagster’s server model enables you to isolate codebases. Problems
```

File: README.md [hunk 12]
```diff
@@ -8,39 +8,29 @@
...
+in one job will not bring down the system or other jobs. Each job can have its own package
+dependencies and Python version.
+- **Debug Jobs from a Rich Interface**: Dagit also includes expansive facilities for understanding
```
Comment (erinkcochran87) [inline README.md:29]:
```suggestion
- **Debug jobs from a rich interface**: Dagit includes expansive facilities for understanding
```

File: README.md [hunk 13]
```diff
@@ -8,39 +8,29 @@
...
+the jobs it orchestrates. When inspecting a run of your job, you can query over logs, discover the
+most time consuming tasks via a Gantt chart, re-execute subsets of steps, and more.
```
Comment (erinkcochran87) [inline README.md:31]:
```suggestion
most time-consuming tasks via a Gantt chart, re-execute subsets of steps, and more.
```

File: README.md [hunk 14]
```diff
@@ -8,39 +8,29 @@
...
+
+## Installation
 
 Dagster is available on PyPI, and officially supports Python 3.6+.
```
Comment (erinkcochran87) [inline README.md:35]:
```suggestion
Dagster is available on PyPI and officially supports Python 3.6+.
```

File: README.md (hunk 3 above)
Comment (yuhan) [reply README.md:48]: I like the idea of having **Getting started** and **Diving in** sections. Look forward to the changes in the docs.

**Review by yuhan — APPROVED** [review]
looks great to me! we can iterate later once we are done with the docs/content improvements

**Pull request merged**